lib/python/rose/apps/rose_bunch.py

```diff
@@ -0,0 +1,458 @@
+        if not multi_args:
+            raise CompulsoryConfigValueError(
+                                        [self.ARGS_SECTION, "command-format"],
+                                        None,
+                                        KeyError(self.ARGS_SECTION))
+
+        self.command_format = conf_tree.node.get_value(
+                                    [self.SETTINGS_SECTION, "command-format"])
+
+        if not self.command_format:
+            raise CompulsoryConfigValueError(
+                                    [self.SETTINGS_SECTION, "command-format"],
+                                     None, KeyError("command-format"))
+
+        # Validate runlists
+        if not self.invocation_names:
+            item, val = sorted(multi_args.items())[0]
+            arglength = len(shlex.split(val.value))
+            self.invocation_names = range(0, arglength)
+        else:
+            arglength = len(self.invocation_names)
+
+        for item, val in sorted(multi_args.items()):
+            if len(shlex.split(val.value)) != arglength:
+                raise InconsistentArglength()
+
+        if conf_tree.node.get_value([self.ARGS_SECTION, "instances"]):
+            raise ConfigValueError([self.ARGS_SECTION, "instances"],
+                                   conf_tree.node.get_value(
+                                    [self.ARGS_SECTION, "instances"]),
+                                   "reserved keyword")
+
+        # Set up instances if needed
+        instances = conf_tree.node.get_value([self.SETTINGS_SECTION, "instances"])
+
+        if instances:
+            try:
+                instances = int(instances)
+                instances = range(instances)
+            except ValueError as err:
+                raise ConfigValueError([self.SETTINGS_SECTION, "instances"],
+                                       instances,
+                                       "not an integer value")
+            if arglength != len(instances):
+                raise InconsistentArglength()
+
+        # Set max number of processes to run at once
+        max_procs = int(conf_tree.node.get_value(
+                                            [self.SETTINGS_SECTION, "limit"]))
+        if max_procs:
+            self.MAX_PROCS = int(max_procs)
+        else:
+            self.MAX_PROCS = arglength
+
+
+        if self.INCREMENTAL:
+            self.dao = RoseBunchDAO(conf_tree)
+        else:
+            self.dao = None
+
+        commands = {}
+        for index, name in enumerate(self.invocation_names):
+            invocation = RoseBunchCmd(name, self.command_format, index)
+            for key, vals in sorted(multi_args.items()):
+                invocation.argsdict[key] = shlex.split(vals.value)[index]
+            if instances:
+                invocation.argsdict["instances"] = instances[index]
+            commands[name] = invocation
+
+        procs = {}
+        log_format = os.path.join(os.environ['ROSE_TASK_LOG_DIR'], "%s")
```

Comment:
I guess this is the only point where this application will work with `rose task-run` and not `rose app-run`? Perhaps it is worth having a logic where if this environment variable is not set, then the logs should go to the current working directory?